src/sage/graphs/matching_covered_graph.py

```diff
@@ -0,0 +1,534 @@
+        Traceback (most recent call last):
+        ...
+        ValueError: input graph is not matching covered
+        sage: # optional - python_igraph
+        sage: import igraph
+        sage: G = Graph(igraph.Graph([(0, 1), (0, 2), (0, 3), (1, 2), (2, 3)]))
+        sage: H = MatchingCoveredGraph(G)
+        Traceback (most recent call last):
+        ...
+        ValueError: input graph is not matching covered
+
+    Providing with a wrong matching::
+
+        sage: G = graphs.CompleteGraph(6)
+        sage: M = Graph(G.matching())
+        sage: M.add_edges([(0, 1), (0, 2)])
+        sage: H = MatchingCoveredGraph(G, matching=M)
+        Traceback (most recent call last):
+        ...
+        ValueError: the input is not a matching
+        sage: N = Graph(G.matching())
+        sage: N.add_edge(6, 7)
+        sage: H = MatchingCoveredGraph(G, matching=N)
+        Traceback (most recent call last):
+        ...
+        ValueError: the input is not a matching of the graph
+        sage: J = Graph()
+        sage: J.add_edges([(0, 1), (2, 3)])
+        sage: H = MatchingCoveredGraph(G, matching=J)
+        Traceback (most recent call last):
+        ...
+        ValueError: the input is not a perfect matching of the graph
+
+    Note that data shall be one of empty or ``None`` or an instance of
+    ``Graph`` or an instance of ``MatchingCoveredGraph``. Otherwise a
+    :class:`ValueError` is returned::
+
+        sage: D = digraphs.Complete(10)
+        sage: D
+        Complete digraph: Digraph on 10 vertices
+        sage: G = MatchingCoveredGraph(D)
+        Traceback (most recent call last):
+        ...
+        ValueError: input data is of unknown type
+    """
+
+    def __init__(self, data=None, matching=None, algorithm='Edmonds',
+                 solver=None, verbose=0, integrality_tolerance=0.001,
+                 *args, **kwds):
+        r"""
+        Create a matching covered graph, that is a connected nontrivial graph
+        wherein each edge participates in some perfect matching
+
+        See documentation ``MatchingCoveredGraph?`` for detailed information.
+        """
+        if kwds is None:
+            kwds = {'loops': False}
+        else:
+            if 'loops' in kwds and kwds['loops']:
+                raise ValueError('loops are not allowed in '
+                                 'matching covered graphs')
+            kwds['loops'] = False
+
+        if data is None:
+            raise ValueError('the graph is trivial')
+
+        elif isinstance(data, MatchingCoveredGraph):
+            Graph.__init__(self, data, *args, **kwds)
+
+        elif isinstance(data, Graph):
+            Graph.__init__(self, data, *args, **kwds)
+            self._upgrade_from_graph(matching=matching, algorithm=algorithm,
+                                     solver=solver, verbose=verbose,
+                                     integrality_tolerance=integrality_tolerance)
+        else:
```

Comment:
remove the `else` and reduce indentation of the `raise`. It's the default action of the method.

Reply:
If I am not mistaken, I suppose, in that case, we need to add `return` at the end of the `if`/ `elif`?

Reply:
I mistakenly suggested that. You are right, the code is ok like that.